x/stake/genesis.go

```diff
@@ -41,7 +44,13 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data types.GenesisState) error
 	}
 
 	keeper.UpdateBondedValidatorsFull(ctx)
-	return nil
+
+	vals := keeper.GetAllValidators(ctx)
```

Comment:
We want all the *bonded* validators, not all validators - I think you want `GetValidatorsBonded` - there may be more than 100 genesis validator candidates